languages/wikipron/src/scrape_and_write.py

```diff
@@ -0,0 +1,123 @@
+        file = open(f"../tsv/{lang}{file_extension}.tsv", "w")
+        count = 0
+        try:
+            for (word, pron) in wikipron.scrape(config):
+                count += 1
+                print(f"{word}\t{pron}", file=file)
+            file.close()
+            return count
+        except (
+            requests.exceptions.Timeout,
+            requests.exceptions.ConnectionError
+        ) as err:
+            print(
+                "Timeout or connection error detected during scrape.",
+                lang, file_extension,
+                str(datetime.now()),
+                err
+            )
+            # Pause execution for 30 min.
+            sleep(1800)
+            retries += 1
+            # Need to close file for it to be truncated when re-opening
+            file.close()
+            pass
+
+    file.close()
+    return None
+
+
+def main():
+    languages_file = open("languages.json", "r")
+    LANGUAGES = json.load(languages_file)
+
+    for iso639_code in LANGUAGES:
+        print(
+            'Currently running:',
+            LANGUAGES[iso639_code]["wiktionary_name"],
+            iso639_code,
+            str(datetime.now())
+        )
+
+        row = [
+            iso639_code, LANGUAGES[iso639_code]["iso639_name"],
+            LANGUAGES[iso639_code]["wiktionary_name"],
+            str(LANGUAGES[iso639_code]["casefold"])
+        ]
+
+        phonemic_config = wikipron.Config(
+            key=iso639_code,
+            casefold=LANGUAGES[iso639_code]["casefold"],
+            no_stress=True,
+            no_syllable_boundaries=True
+        )
+        phonetic_config = wikipron.Config(
+            key=iso639_code,
+            casefold=LANGUAGES[iso639_code]["casefold"],
+            phonetic=True,
+            no_stress=True,
+            no_syllable_boundaries=True
+        )
+
+        phonemic_count = call_scrape(iso639_code, phonemic_config, "_phonemic")
+        # Skip phonetic if phonemic failed to complete scrape
+        if phonemic_count is not None:
+            phonetic_count = call_scrape(
+                iso639_code,
+                phonetic_config,
+                "_phonetic"
+            )
+
+        phonemic_path = f"../tsv/{iso639_code}_phonemic.tsv"
+        phonetic_path = f"../tsv/{iso639_code}_phonetic.tsv"
+        # Remove files for languages that failed to be scraped
+        # within set amount of retries.
+        if phonemic_count is None or phonetic_count is None:
+            print(f"TOO MANY RETRIES ON {iso639_code} MOVING ON TO NEXT.")
+            if os.path.exists(phonemic_path):
+                os.remove(phonemic_path)
+            if os.path.exists(phonetic_path):
+                os.remove(phonetic_path)
+            continue
+        # Remove files for languages that failed to call scrape altogether
+        # or for which wikipron returned nothing
+        elif phonemic_count == 0 and phonetic_count == 0:
+            print(f"FAILED TO SCRAPE {iso639_code}.")
```

Comment:
This was the case for Cantonese in #57, correct?

Reply:
That's correct, as well as for Khmer and Thai. Both `phonetic_count` and `phonemic_count` are 0.